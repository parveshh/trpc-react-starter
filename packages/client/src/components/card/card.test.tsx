import React from 'react';
import { Card } from '.';
import { render } from '@testing-library/react';

describe('Card Tests', () => {
  test('should render', async () => {
    const card = render(
      <Card icon={<div />} title='Heading'>
        test
      </Card>
    );
    expect(card).toBeTruthy();
    const heading = await card.findByRole('heading');
    expect(heading).toBeTruthy();
    expect(heading?.textContent).toBe('Heading');
    const content = await card.findByRole('contentinfo');
    expect(content).toBeTruthy();
    expect(content?.textContent).toBe('test');
  });
});
