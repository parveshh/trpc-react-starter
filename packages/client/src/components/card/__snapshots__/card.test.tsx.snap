// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Tests should render 1`] = `
<Card
  icon={<div />}
  title="test"
>
  test
</Card>
`;
